lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb

```diff
@@ -605,15 +704,22 @@ def self.add_to_destination(api_token, owner_name, app_name, release_id, destina
         UI.message("DEBUG: POST #{url}") if ENV['DEBUG']
         UI.message("DEBUG: POST body #{JSON.pretty_generate(body)}\n") if ENV['DEBUG']
 
-        response = connection.post(url) do |req|
-          req.headers['X-API-Token'] = api_token
-          req.headers['internal-request-source'] = "fastlane"
-          req.body = body
+        status, message, response = retry_429_and_error do 
+          response = connection.post(url) do |req|
+            req.headers['X-API-Token'] = api_token
+            req.headers['internal-request-source'] = "fastlane"
+            req.body = body
+          end
         end
 
-        UI.message("DEBUG: #{response.status} #{JSON.pretty_generate(response.body)}\n") if ENV['DEBUG']
-
-        case response.status
+        case status
+        when 0, 429
+          if status == 0
+            UI.error("Faraday http exception adding to destination: #{message}")
+          else
+            UI.error("Retryable error adding to destination: #{status}: #{message}")
+          end
+          false
```

Comment:
I would move repeating code to a separate method and reuse it. Also I think `when 0, 429` looks better if we split it to `when 0... when 429...`

Reply:
We discussed it.